src/rust/engine/src/nodes.rs

```diff
@@ -76,15 +70,16 @@ impl StoreFileByDigest<Failure> for Context {
 /// NodeKey's impl of graph::Node handles the rest.
 ///
 /// The Item type of a WrappedNode is bounded to values that can be stored and retrieved
-/// from the NodeResult enum. Due to the semantics of memoization, retrieving the typed result
-/// stored inside the NodeResult requires an implementation of TryFrom<NodeResult>. But the
+/// from the NodeOutput enum. Due to the semantics of memoization, retrieving the typed result
```

Comment:
NB: I renamed `NodeResult` to `NodeOutput` (which was never a "result" in the rust sense) and reused the name `NodeResult` for an actual `Result` type alias.